dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -35,4 +35,4 @@ properties:
         adequately chosen. On storage backends with uniform erase-blocks it
         should be at least the erase-block-size, on storage backends with
         non-uniform erase-blocks it should be at least the largest
-        erase-block-size.
+        erase-block-size. The size can be 0 if partition is read-only.
```

Comment:
Would not be "The cache-size property will be ignored if the partition is read-only" more correct?

Reply:
I wouldn't classify it as "more correct" as basically both are true. That is "The size can be 0 if partition is read-only" and "The cache-size property will be ignored if the partition is read-only" refer to different things.

Reply:
I agree with @jfischer-no here. I would like to see this documentation updated, because when the partition is ready only, the driver will effectively ignore the `cache-size` property by multiplying by by zero. Also, it looks like the cache implementation only works on uniform erase blocks (unless I'm missing something), so should we drop the prior sentence about non-uniform erase blocks?

Reply:
Flashdisk never worked in read-write mode on non-uniform erase blocks partitions. Prior to `zephyr,flash-disk` compatible it could be used assuming the user prevented writes in application (if not, some data would simply be lost). With changes in this PR, with the default Kconfig, at least the user will get an error if such flashdisk is not read-only.

Is is technically possible to implement the non-uniform erase blocks support, but it is not done because there is no easy way to test it. The simulated flash does not support non-uniform pages and I don't have any such device.

Reply:
Now the description includes both facts, i.e. the size can be 0 if read-only and that the size is ignored when read-only.

Reply:
Actually being ignored somewhat implies that it can be 0, so I'll delete the can be 0.